src/js/view/view.js

```diff
@@ -627,8 +583,8 @@ function View(_api, _model) {
             return !!(fsElement && fsElement.id === _model.get('id'));
         }
         // if player element view fullscreen not available, return video fullscreen state
-        return _instreamModel ? _instreamModel.getVideo().getFullScreen() :
-            _model.getVideo().getFullScreen();
+        const provider = _model.getVideo();
```

Comment:
Can we do `model.get('provider')` here? Would like to remove the getVideo api

Reply:
That will return `{ name: 'provider' }`. The provider's fullscreen state, should update the model. Then we can remove this.

There is also another place in the view where we set fullscreen on the provider (iOS) if DOM fullscreen is not available. I would like to do that without accessing the provider in the view. Maybe a view-model event that doesn't get forwarded to the API (view events do get forwarded by API).

Reply:
@johnBartos `model.get('provider')` currently returns an object with the provider name (eg. `{name: "html5"}`), which is what gets returned when the public api calls `jwplayer().getProvider()`. This would result in a breaking change to the public api.